### PR TITLE
add new properties to the swagger spec

### PR DESCRIPTION
**Pull request by eshaparmar**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ravbhatnagar** [comment]
Adding new properties requires a new api-version. So, if needs to be done in the same api-version for some reason , it will need approval from Azure API review board ( Adrian Hall, Jeffery Richter, Johan Stenberg etc.)

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json
Comment (nitinbps) [inline specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json:1032]: >report [](start = 71, length = 6)

probably read-only

**Review by nitinbps — APPROVED** [review]
:shipit:

**Comment by eshaparmar** [comment]
I have created a new version for this change

**Review by yungezz — COMMENTED** [review]
LGTM. waitting for arm review since new api version

**Review by yungezz — APPROVED** [review]

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (pilor) [inline specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:4]: >2020-06-25 [](start = 16, length = 10)

please make the first commit the old api-versions content in the new path then start making changes to make these easier to review in the future

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/examples/createOrUpdateGuestConfigurationAssignment.json
Comment (pilor) [inline specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/examples/createOrUpdateGuestConfigurationAssignment.json:37]: >null [](start = 14, length = 4)

the examples need to be cleaned up.  "id" can never be null for example

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (pilor) [inline specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:998]: these both come from the resource ID and should not be duplicated within the resource

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (pilor) [inline specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:1031]: >AssignmentReport [](start = 33, length = 16)

why are there different definitions for assignmentReport?  Whats the difference between them?

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (pilor) [inline specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:1030]: >latestAssignmentReport [](start = 9, length = 22)

pulling the content of children into a parent is something that should be avoided unless absolutely necessary.  It breaks RBAC since users without /read permissions on /reports will now be able to see the report in the guestConfigAssignment

**Review by pilor — CHANGES_REQUESTED** [review]
:clock1:

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (eshaparmar) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:998]: these new properties are needed by the ARG team. I can remove the subscriptionId bt targetResourceId is being used for queries.

Comment (eshaparmar) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:998]: for the purpose of establishing the correlation id's - For cross dataset references/ joins. since guest config reosurce is a proxy resource and the other data set is a tracked - arg team cant do queries based on the default resource ID. so they asked us to add this property so they can use this for joins and stuff

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (eshaparmar) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:1031]: removed it - that wasn't needed

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (eshaparmar) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:1030]: these new properties are required to provide a complete data set for a guest configuration assignment resource. this would be sent to ARG and for guest config customers/partners to query. I do understand your concern wrt RBAC but i cant think of a scenario where a user would have access to guest config resource but not its child resource - report in this case. As all the relevant compliance information is in the report.
having said that - i am open to other suggestions that you may have to achieve the same.

Comment (eshaparmar) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:1024]: I was thinking about the RBAC thing more - and in the arm manifest we only list one proxy type which is guestConfigurationAssignment type. We dont specify reports as a child type. so i am wondering how would someone be able to place RBAC rules on the reports when its not actually a type? Am i missing something here?


"resourceTypes": [
        {
            "name": "guestConfigurationAssignments",
            "routingType": "ProxyOnly, Extension",
            "resourceDeletionPolicy": "Cascade",
            "linkedOperationRules": [],
            "quotaRule": {
                "quotaPolicy": "None"
            }

Comment (pilor) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:1024]: RBAC is based on your operations API:

Operation         : Microsoft.GuestConfiguration/guestConfigurationAssignments/reports/read
OperationName     : Microsoft.GuestConfiguration/guestConfigurationAssignments/reports/read
ProviderNamespace : Microsoft Guest Configuration
ResourceName      : Microsoft.GuestConfiguration/guestConfigurationAssignments
Description       : Get guest configuration assignment report.
IsDataAction      : False

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (pilor) [inline specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:919]: doesn't the "id" show up 3 times now?  The guestConfigAssignment has a "latestReportId" then this has a resource ID (which has the ID at the end of it) and a separate reportId property.  Why do we need all 3?

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (pilor) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:1024]: Since its all readonly and not modifiable at either level I won't block on it but this is pretty messy just to get ARG queries to work

**Review by pilor — APPROVED** [review]
Please prune down the duplicated info between the assignment and the embedded report

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json
Comment (eshaparmar) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json:919]: i think there is some confusion  here :
"reportId": "7367cbb8-ae99-47d0-a33b-a283564d2cb1"
"id": "/subscriptions/mysubscriptionid/resourceGroups/myResourceGroupName/providers/Microsoft.Compute/virtualMachines/myvm/providers/Microsoft.GuestConfiguration/guestConfigurationAssignments/AuditSecureProtocol/reports/7367cbb8-ae99-47d0-a33b-a283564d2cb1",
targetResourceId == VM resourceID

only duplicate here is the latestReportId in the guestCOnfigurationAssignment resource at the top level and then the latestReport - would also have it. This is ok because we are not populating the latestReport object at all times right now. We are only sending it to ARG, api calls to control plane will have latestReport as null.

**Pull request merged**
